tests/integration/targets/test_gcs_backend/tasks/main.yml

```
@@ -0,0 +1,109 @@
+        src: 'main.tf.j2'
+        dest: "{{ tmpdir.path }}/main.tf"
+
+    - name: Apply Terraform configuration
+      cloud.terraform.terraform:
+        state: present
+        project_path: "{{ tmpdir.path }}"
+        force_init: true
+
+    - name: Download the state file from storage bucket
+      google.cloud.gcp_storage_object:
+        action: download
+        bucket: "{{ test_gcs_backend_bucket_name }}"
+        src: "{{ test_gcs_backend_object_prefix }}/default.tfstate"
+        dest: "{{ tmpdir.path }}/content.tfstate"
+        project: "{{ GCP_PROJECT_ID }}"
+        auth_kind: "serviceaccount"
+        service_account_file: "{{ GCP_SERVICE_ACCOUNT_CREDENTIALS_FILE }}"
+      register: download_state_file
+
+    - name: Check if downloaded file exists at destination
+      stat:
+        path: "{{ tmpdir.path }}/content.tfstate"
+      register: file_status
+
+    - name: Ensure downloaded state file is not empty
+      ansible.builtin.assert:
+        that:
+          - download_state_file is changed
+          - download_state_file is not failed
+          - file_status.stat.exists is true
+          - file_status.stat.size > 0
+          - file_status.stat.mimetype == "application/json"
+
+    # Delete the Backend resources
+    # currently, fails as google.cloud.gcp_storage_bucket module does not support force delete non-empty bucket
+    # object deletion does not help as buckets with versioning enabled retain the deleted objects
+    - name: Delete GCS backend resources
+      ansible.builtin.include_role:
+        name: cloud.terraform_ops.gcs_backend
+      vars:
+        gcs_backend_operation: delete
+        gcs_backend_bucket_name: "{{ test_gcs_backend_bucket_name }}"
+        gcs_backend_gcp_project: "{{ GCP_PROJECT_ID }}"
+        gcs_backend_auth_kind: serviceaccount
+        gcs_backend_service_account_file: "{{ GCP_SERVICE_ACCOUNT_CREDENTIALS_FILE }}"
+      register: delete_bucket_result
+      ignore_errors: true
```

Rather than ignoring all errors here, I think we should define `failed_when` so that it will fail if the error is anything other than `'The bucket you tried to delete is not empty.'`

---

removed `ignore_errors` from tests, added to role task for delete operations
this will not `fail` the tests and at the same time display error message when role is ran to notify user to delete manually

---

I think we do have to ignore_errors here, sorry for the confusion! I was trying to figure out a way to only ignore the error we are expecting but apparently `failed_when` doesn't work with role includes.

---

adding `ignore_errors` in the test task below would still cause tests to fail as this task in roles/gcs_backend/tasks/delete.yml https://github.com/redhat-cop/cloud.terraform_ops/pull/3/files#diff-266ad447cb12fecdb0ca92156ec232d1a9aac87e9b8878c750ac9ee182997419R12-R18 does not `ignore` errors (error is expected due to being unable to delete non-empty buckets)
```
    - name: Delete GCS backend resources
      ansible.builtin.include_role:
        name: cloud.terraform_ops.gcs_backend
      vars:
        gcs_backend_operation: delete
        ...
```
are we okay with this behavior in tests? also `ignore_errors` does not work on `include_role` if task inside the role fails, it fails